hs/src/Reach/Eval/Core.hs

```diff
@@ -3119,11 +3119,9 @@ evalApplyClosureVals clo_at (SLClo mname formals (JSBlock body_a body _) SLCloEn
       let mlvls = map fst msvs
       let mvs = map snd msvs
       let lvl = mconcat $ xlvl : mlvls
-      -- Note: This test might be too expensive, so try it
-      -- chicken-fork and many other fork-ing programs would be improved by
-      -- this when all pays are zero, but it doesn't work because ints have a
-      -- srcloc
-      let all_same = False && all (== xv) mvs
+      -- We check equivalence here with Equiv typeclass. it's shallow checking but performant
+      let all_same = all (equiv xv) mvs
+      -- let all_same = False && all (equiv xv) mvs
       case all_same of
         True -> no_prompt $ (lvl, xv)
```

Comment:
I realized recently that it is actually dangerous to remove the prompt in this situation (the case in `no_prompt`) because the final return might be at the end where there was an early return. The key point then is to (a) return the `xv` and (b) include the prompt in the dynamic code

Reply:
I opened an issue for this one since I'm not quite sure how to accomplish that: https://trello.com/c/fLRQSHyq/1471-fix-noprompt-safety-in-core

Reply:
Make a function called something like `no_prompt'` that does the second half of `no_prompt`. Have `no_prompt` call it and have this `all_same` case call `no_promp'`. That way we will only ever remove the prompt if there was a single return